### ISSUE TITLE
builtins no longer pipe-able (do not terminate)

### ISSUE THREAD

**Issue by con-f-use** [xonsh/xonsh#3476]
I can no longer pipe the output of built-in xonsh commands such as `xonfig` or `history` into other commands. They simply produce no output and run forever.

e.g.:

```
$ echo hallo
hallo

$ history
echo hallo

$ history | grep al     # prints nothing and never terminates

$ xonfig | grep xonsh    # the same
```

## xonfig

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.9.14               |
| Git SHA          | 5648b8d5             |
| Commit Date      | Oct 17 19:02:06 2019 |
| Python           | 3.6.9                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.9                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | ubuntu               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

</details>

**Comment by scopatz** [comment]
Hmm that is not good.  I wonder what is happening here.  I can't reprodude this:

```sh
scopatz@artemis ~ $ echo hallo                                                                                                                                          
hallo
scopatz@artemis ~ $ history | grep al                                                                                                                                   
echo hallo
```

**Comment by gforsyth** [comment]
Hey @con-f-use -- can you try updating `xonsh`?  Your version shows 0.9.14 but the git sha in your `xonfig` is from just after we merged the job control fixes and there were some small piping issues in those.  We've since fixed them (I think!) so it's worth a reinstall to make sure.

**Comment by con-f-use** [comment]
Still a problem for me on 177e1876

**Comment by con-f-use** [comment]
Disabling the `bashisms` plugin resolves this problem.

**Comment by anki-code** [comment]
Closed with reference to #4250
